holidays/holiday_groups.py

```diff
@@ -804,27 +842,47 @@ def _add_eid_al_fitr_day_four(self, holiday_name) -> Set[date]:
         https://en.wikipedia.org/wiki/Eid_al-Fitr
         """
         return self._add_islamic_calendar_holiday(
-            holiday_name, "EID_AL_FITR", days_delta=+3
+            holiday_name,
+            self._islamic_calendar.eid_al_fitr_dates(self._year),
+            days_delta=+3,
+        )
+
+    def _add_hari_hol_johor(self, holiday_name) -> Set[date]:
+        """
+        Hari Hol Johor.
+
+        https://publicholidays.com.my/hari-hol-almarhum-sultan-iskandar/
+        """
+        return self._add_islamic_calendar_holiday(
+            holiday_name,
+            self._islamic_calendar.hari_hol_johor_dates(self._year),
         )
 
     def _add_islamic_calendar_holiday(
-        self, holiday_name, hol_type: str, days_delta: int = 0
+        self,
+        name: str,
+        dates: Iterable[Tuple[date, bool]],
+        days_delta: int = 0,
     ) -> Set[date]:
         """
         Add lunar calendar holiday.
-        """
-        dates = set()
-        years = (
-            (self._year - 1, self._year) if days_delta > 0 else (self._year,)
-        )
-        for year in years:
-            for dt in _IslamicLunar.islamic_holiday_date(year, hol_type):
-                if days_delta != 0:
-                    dt += td(days=days_delta)
-                dt = self._add_holiday(holiday_name, dt)
-                if dt:
-                    dates.add(dt)
-        return dates
+
+        Appends customizable estimation label at the end of holiday name if
+        holiday date is an estimation.
+        """
+        added_dates = set()
+        estimated_label = getattr(self, "estimated", "estimated")
```

Comment:
In Pakistan (only use for now), this attribute is named `estimated_label`, need to synchronize :)